src/libpriv/rpmostree-diff.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; either version 2 of the licence or (at
+ * your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General
+ * Public License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place, Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#pragma once
+
+#include <string>
+#include <exception>
+#include <sstream>
+#include <memory>
+
+#include <ostree.h>
+
+#include "rust/cxx.h"
+
+namespace rpmostreecxx {
+
+class RPMDiff final {
```

Comment:
Ohhh, a class encapsulating a GPtrArray. That might be a way out for https://github.com/coreos/rpm-ostree/pull/2568 until we get native pointer support in cxx.rs.

Reply:
Yeah, I think so; though in this case I wanted to add methods.  I need to study that one a bit more, there may be a nicer way.  But yeah, an `RpmOstreePackageArray` wouldn't be too bad to start maybe?